src/utils/index.js

```diff
@@ -49,3 +50,33 @@ export const filteredPath = (resource, language, topic) => {
     stringValueOrAll(language)
   )}/topic/${toSlug(stringValueOrAll(topic))}`;
 };
+
+/**
+ * Returns a shuffled copy of the array using the Fisher-Yates algorithm.
+ *
+ * @param {any[]} array Array to be copied and shuffled
+ */
+export const shuffleCopy = (array) => {
+  const copy = [...array];
+  for (let i = copy.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    const temp = copy[i];
+    copy[i] = copy[j];
+    copy[j] = temp;
+  }
+  return copy;
+};
+
+/**
+ * This hook allows us to know whether we're in SSR phase or in client hydration
+ * phase.
+ *
+ * @return booleans indicating whether we're in Server Side Rendering or
+ * Client Side Rendering
+ * @see {@link https://blog.logrocket.com/fixing-gatsbys-rehydration-issue/}
+ */
+export const useSSR = () => {
+  const [isServer, setIsServer] = useState(true);
+  useEffect(() => setIsServer(false), []);
+  return { isServer, isClient: !isServer };
+};
```

Comment:
I think it would make more sense to call this hook something like `isFirstRender`, since it will render true for the first render on the server and for the first render on the client too. So something like the `isClient` boolean will confuse people because it will be `false` on the first render in the client. I propose renaming the hook to `isFirstRender` and then just returning a single boolean.

Reply:
You're absolutely right! I myself got confused with how many times this value was true ^^ It's much clearer now. I renamed it to `useIsFirstRender`, starting with `use` to make it clear that this is a custom hook. Since this function encapsulates a hook, [rules of hooks](https://reactjs.org/docs/hooks-rules.html) apply to it (see [this page](https://reactjs.org/docs/hooks-custom.html) for more info).